frontend/src/components/pages/FridgeManagement/FridgeCheckIns/EditCheckin.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+  Button,
+  Center,
+  CloseButton,
+  Container,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Input,
+  Spinner,
+  Text,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { format, isAfter, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import { CreateCheckInFields } from "../../../../types/CheckInTypes";
+import ErrorMessages from "./ErrorMessages";
+
+const EditCheckInPage = (): JSX.Element => {
+  const { id } = useParams<{ id: string }>();
+  const [date, setDate] = useState<string>("");
+  const [startTime, setStartTime] = useState<string>("");
+  const [endTime, setEndTime] = useState<string>("");
+  const [notes, setNotes] = useState<string | undefined>("");
+  const [timeRangeError, setTimeRangeError] = useState("");
+  const toast = useToast();
+  const history = useHistory();
+
+  const getCheckInData = async () => {
+    const checkInResponse = await CheckInAPIClient.getCheckInsById(id);
+    setDate(checkInResponse.startDate);
+    setStartTime(format(new Date(checkInResponse.startDate), "kk:mm"));
+    setEndTime(format(new Date(checkInResponse.endDate), "kk:mm"));
+    setNotes(checkInResponse.notes);
+  };
+
+  const validateForm = () => {
+    let newError = "";
+    let valid = true;
+    if (
+      isAfter(
+        parse(startTime, "kk:mm", new Date()),
+        parse(endTime, "kk:mm", new Date()),
+      )
+    ) {
+      valid = false;
+      newError = ErrorMessages.endTimeBeforeStartTime;
+    }
+
+    setTimeRangeError(newError);
+    return valid;
+  };
+
+  const onSaveClick = async () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    const checkInData: CreateCheckInFields = {
+      startDate: parse(startTime, "kk:mm", new Date(date)).toString(),
+      endDate: parse(endTime, "kk:mm", new Date(date)).toString(),
+      notes,
+    };
+
+    const res = await CheckInAPIClient.updateCheckInById(id, checkInData);
+    if (!res) {
+      toast({
+        title: "Checkin could not be updated. Please try again",
+        status: "error",
+        duration: 7000,
+        isClosable: true,
+      });
+      return;
+    }
```

Comment:
This also makes me wonder that this toast logic is not working cuz i was seeing success toast, but should have seen error toast. This could be cuz the UpdateCheckById returns `error as Checkin` on failure so we prolly need some other check to ensure that

Reply:
is there a reason we return the error as `Checkin` and not `false`?

Reply:
i think the idea was so that the return type is the same. However if returning as `Checkin | boolean` works and we dont encounter any type issues (dont think we will), then yeah lets do this change.